wdl2cwl/main.py

```diff
@@ -113,52 +96,11 @@ def get_cwl_docker_requirements(
     return cwl.DockerRequirement(dockerPull=dockerpull)
 
 
-def get_literal_name(
-    expr: Union[
-        WDL.Expr.Boolean,
-        WDL.Expr.Int,
-        WDL.Expr.Float,
-        WDL.Expr.Array,
-    ],
-) -> str:
-    """Translate WDL Boolean, Int, Float, or Array Expression."""
-    # if the literal expr is used inside WDL.Expr.Apply
-    # the literal value is what's needed
-    parent = expr.parent  # type: ignore[union-attr]
-    if isinstance(parent, WDL.Expr.Apply):
-        return expr.literal.value  # type: ignore
-    raise WDLSourceLine(expr, ConversionException).makeError(
-        f"The parent expression for {expr} is not WDL.Expr.Apply, but {parent}."
-    )
-
-
-def get_expr_name(wdl_expr: WDL.Expr.Ident) -> str:
-    """Extract name from WDL expr."""
-    if not hasattr(wdl_expr, "name"):
-        raise WDLSourceLine(wdl_expr, ConversionException).makeError(
-            f"{type(wdl_expr)} has not attribute 'name'"
-        )
-    return get_input(wdl_expr.name)
-
-
-def get_expr_name_with_is_file_check(wdl_expr: WDL.Expr.Ident) -> str:
-    """Extract name from WDL expr and check if it's a file path."""
-    if wdl_expr is None or not hasattr(wdl_expr, "name"):
-        raise WDLSourceLine(wdl_expr, ConversionException).makeError(
-            f"{type(wdl_expr)} has not attribute 'name'"
-        )
-    expr_name = get_input(wdl_expr.name)
-    is_file = isinstance(wdl_expr.type, WDL.Type.File)
-    return expr_name if not is_file else f"{expr_name}.path"
-
-
 class Converter:
     """Object that handles WDL Workflows and task conversion to CWL."""
 
     def __init__(self) -> None:
-        """Initialize the sets used by the object and prevent inconsistent behaviours."""
-        self.non_static_values: Set[str] = set()
-        self.optional_cwl_null: Set[str] = set()
+        self.ctx = ConversionContext()
```

Comment:
Note to self: Probably good idea to reset the context (clear set) when the file was written to the output stream (i.e. when conversion has finished), to avoid stale state...

Reply:
And made it into a dataclass so I could pass it around as a single variable.